### PR TITLE
Remove optionals from styling interfaces

### PR DESCRIPTION
**Pull request by phkuo**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Some quality of life changes leveraging partial interfaces. Consumers can use partials to define overrides, while the package can use interfaces without optionals so the compiler can verify all expected values are being defined.

### CODE REVIEW

File: packages/styling/src/interfaces/ITheme.ts [hunk 1]
```diff
@@ -5,5 +5,10 @@ import { ISemanticColors } from './ISemanticColors';
 export interface ITheme {
   palette: IPalette;
   fonts: IFontStyles;
-  semanticColors?: ISemanticColors;
+  semanticColors: ISemanticColors;
 }
+export interface IPartialTheme {
+  palette: Partial<IPalette>;
```
Comment (dzearing) [inline packages/styling/src/interfaces/ITheme.ts]: Why is this not optional too?

Comment (dzearing) [inline packages/styling/src/interfaces/ITheme.ts]: I think you should be able for example to provide just fonts.

Comment (phkuo) [inline packages/styling/src/interfaces/ITheme.ts]: You're right. I had done this before I pulled change that added a default palette.

File: packages/styling/src/styles/theme.ts [hunk 2]
```diff
@@ -38,27 +40,29 @@ export function getTheme(): ITheme {
...
-export function createTheme(theme: Partial<ITheme>): ITheme {
-  let newPalette = { ..._theme.palette, ...theme.palette };
+export function createTheme(theme: IPartialTheme): ITheme {
+  let newPalette = { ...DefaultPalette, ...theme.palette };
 
   return {
     palette: newPalette,
     fonts: {
-      ..._theme.fonts,
+      ...DefaultFontStyles,
```
Comment (dzearing) [inline packages/styling/src/styles/theme.ts]: I don't know why you changed this. Technically, _theme.fonts should be whatever was previously set. Why reset back here? This change doesn't make sense to me.

**Pull request merged**